packages/event-handler/src/rest/BaseRouter.ts

```diff
@@ -185,14 +195,24 @@ abstract class BaseRouter {
         throw new NotFoundError(`Route ${path} for method ${method} not found`);
       }
 
-      const result = await route.handler.apply(options?.scope ?? this, [
+      const handler =
+        options?.scope != null
+          ? route.handler.bind(options.scope)
+          : route.handler;
+
+      const middleware = composeMiddleware([...this.middlwares]);
+
+      const result = await middleware(
         route.params,
         {
           event,
           context,
           request,
         },
-      ]);
+        () => handler(route.params, { event, context, request })
```

Comment:
Note here that we are just treating the handler like another middleware that doesn't call next